src/internal/components/chart-series-details/index.tsx

```diff
@@ -19,14 +19,14 @@ export interface ChartSeriesDetailsProps extends BaseComponentProps {
   details: ReadonlyArray<ChartSeriesDetailItem>;
 }
 
-export default memo(ChartSeriesDetails);
+export default memo(forwardRef(ChartSeriesDetails));
 
-function ChartSeriesDetails({ details, ...restProps }: ChartSeriesDetailsProps) {
+function ChartSeriesDetails({ details, ...restProps }: ChartSeriesDetailsProps, ref?: React.Ref<HTMLDivElement>) {
```

Comment:
What is happening here? Why do you make this change, but there is no any use of this ref?

Also, how can `ref` argument be undefined? Isn't it always supplied by react?

Reply:
> What is happening here? Why do you make this change, but there is no any use of this ref?

This ref is used [here](https://github.com/cloudscape-design/components/pull/1751/files#diff-3828e76418234b3e69ed5fee63c4b82beac8da7d413fc71047088ac70ce00120R127). Before, we could pass the element as child of the live region but now that there might be interactive elements, we need to do it with a ref and pass it to the live region as `source`.

> how can ref argument be undefined? Isn't it always supplied by react?

Right, changed [here](https://github.com/cloudscape-design/components/pull/1751/commits/058739dc4d86cef8a781aac06343b40a3562280d).